react.md

```diff
@@ -10,37 +10,286 @@ mainmenu: true
 
 We implement a standard set of React libraries and tools in order to:
 
-- reduce onboarding time for developers
-- reduce complexity & time required for context switching between projects
-- increase number of developers who can assist on more projects, if developers know the same libraries and tools
+- Reduce onboarding time for new developers.
+- Reduce complexity & time required for context switching between projects.
+- Ensure everyone is contributing to a common knowledge base.
 
 ### Other Choices?
-If considering implementing another solution than our standard approach below, in a section, we recommend clearing it by one of the team leads. There may be an existing solution that most of our team will already know how to implement, build, debug and help with.
 
-## Node
+This guide is more of a strong recommendation than a strict mandate. We recognize that every project is different and some offer particularly unique requirements and technical challenges. That said, deviating from our best practices needs to be justified by a solution below not meeting our needs, so please discuss your idea with your project lead developer or a senior JS developer before implementation. There may be existing solutions that most of our team will already know how to implement and be able to build, debug and help with.
 
-### NVM
-- TODO: Documentation, I think this is our best solution?
-- TODO: How to setup .nvmrc
+## Setting up your workstation
+
+### Node + NVM (Node Version Manager)
+You should install node with [nvm](https://github.com/nvm-sh/nvm), not with Homebrew or the nodejs.org installer. If you've already installed node, please uninstall it before installing nvm. nvm makes it easy to run multiple versions of node side by side on the same machine, allowing you to switch from project to project without running into version issues.
+
+#### .nvmrc
+Each React project should include an [`.nvmrc`](https://github.com/nvm-sh/nvm#nvmrc) with the version of node the project expects.
 
 ### Package management
-Use `npm`, this is a solved problem
+All third-party dependencies should be installed with `npm`. Avoid copying & pasting third-party code directly into the project and avoid using other front-ends to the npm registry like yarn.
+
+All projects should include a `package-lock.json` file to ensure builds are reliable from machine to machine. Please check this file into version control, and avoid deleting & regenerating it. If you run into an issue that you can only solve by regenerating `package-lock.json`, please consult a senior JS developer before doing so to make sure it's absolutely necessary.
```

Comment:
love this part. just because the deletion of `package-lock` has caused a lot of frustration in the past!